### DISCUSSION TITLE
invisible watermark plugin

### DISCUSSION THREAD

**Discussion by harishinka** [obsproject/obs-studio#7402] [Development Questions] [answered]
Hi,
I wanted to know if its possible to integrate invisible watermark library as plugin for live streaming . Basically I am looking at following sequence:
Camera input->Apply invisible watermark to frames-> Encode->Stream (HLS DASH).
I have got the library for invisible watermark developed in Cplusplus and wanted to know if it is possible to hook up before encoding process
If the watermark is applied after encode, then the stream needs to decode, apply watermark and encode which will add delay in streaming. Hence wanted to apply watermark before encode .

**Comment by cristian44137** [comment]
Why don't you just create another scene, drag the camera there in fullscreen and add the watermark? Then you would add that scene as a source in any other scene.

**Comment by Fenrirthviti** [comment, marked as answer]
Please use the [ideas page](https://ideas.obsproject.com/) for feature requests. Make sure you read any [pinned posts](https://github.com/obsproject/obs-studio/discussions/5206) on discussion boards before opening a new discussion.

Thank you!
